### ISSUE TITLE
0.9.0.1 tools bundle missing hybi/stream_reader.js

### ISSUE THREAD

**Issue by sdarnell** [sdarnell/meteor-win#57]
When building the windows:meteor-tool pakage, the generated bundle is missing hybi/stream_reader.js buried deep in the package tree.

My guess is that there is some obscure npm/node/node-tar issue with long filenames, but I haven't pinpointed it yet. 

Temporarily I repackaged the bootstrap package to include it, but the tool package uploaded to the package server has the file missing.

Here's the exception that gets thrown for reference:

```
c:\meteor\test2>meteor --release windows:METEOR@0.9.0.1-rc3 search windows

C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\main.js:1296
}).run();
   ^
Error: Cannot find module './hybi/stream_reader'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\unipackages\livedata\npm\node_modules\faye-websocket\node_modules\websocket-driver\lib\websocket\driver\hybi.js:4:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\unipackages\livedata\npm\node_modules\faye-websocket\node_modules\websocket-driver\lib\websocket\driver\client.js:5:18)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\unipackages\livedata\npm\node_modules\faye-websocket\node_modules\websocket-driver\lib\websocket\driver.js:7:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\unipackages\livedata\npm\node_modules\faye-websocket\lib\faye\websocket.js:8:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object._.extend.Npm.require (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\bundler.js:1095:22)
    at _.extend._launchConnection (packages/livedata/stream_client_nodejs.js:121)
    at new LivedataTest.ClientStream (packages/livedata/stream_client_nodejs.js:28)
    at new Connection (packages/livedata/livedata_connection.js:52)
    at Object.DDP.connect (packages/livedata/livedata_connection.js:1538)
    at new ServiceConnection (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\service-connection.js:48:42)
    at openPackageServerConnection (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\package-client.js:32:10)
    at Object.exports.updateServerPackageData (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\package-client.js:193:16)
    at _.extend._refresh (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\catalog.js:189:40)
    at _.extend.refresh (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\catalog.js:153:14)
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\commands-packages.js:91:22
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\commands-packages.js:80:11
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\buildmessage.js:186:11
    at _.extend.withValue (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\fiber-helpers.js:112:14)
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\buildmessage.js:183:27
    at _.extend.withValue (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\fiber-helpers.js:112:14)
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\buildmessage.js:181:16
    at _.extend.withValue (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\fiber-helpers.js:112:14)
    at Object.capture (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\buildmessage.js:172:21)
    at exports.doOrDie (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\commands-packages.js:79:31)
    at refreshOfficialCatalogOrDie (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\commands-packages.js:90:3)
    at main.registerCommand.name [as func] (C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\commands-packages.js:936:3)
    at C:\Users\stephen\AppData\Local\.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\main.js:1255:23
```

**Comment by sdarnell** [comment]
Shortening the path seems to fix things ... npm issue?

```
mkdir c:\t
set TEMP=c:\t
```

before building the tarballs

**Comment by peterennis** [comment]
This is 128 characters:
C:\Users\stephen\AppData\Local.meteor\packages\windows_meteor-tool\1.0.27\meteor-tool-os.windows.x86_32\tools\package-client.js

Here is a reference to 127 character limit:
Command-Line Filenames Appear Limited to 127 Characters
It dates back to Win95 but: Article ID: 121059 - **Last Review: October 26, 2013** - Revision: 2.0
http://support.microsoft.com/kb/121059

I periodically run into path and environment length limitations.
Maybe its time for moving the build to Powershell?

**Comment by sdarnell** [comment]
Yes it is a path length issue, and may be a 127 character limit, but switching to PowerShell won't help because it is in the area where we execute npm directly from node.
I think the KB article refers to pre-WinXP so I don't think it still applies - but there is some software in between that has an issue with short paths (127 is short :( ).

**Comment by peterennis** [comment]
Some more research:
http://visualstudio.uservoice.com/forums/121579-visual-studio/suggestions/4954037-fix-260-character-file-name-length-limitation

There is a good reference in the comments that may help:
Colin Bowern  commented  ·  July 22, 2014 9:35 PM
**I think you need to start chipping away at this problem. Node.js re-introduces this as well:**
http://stackoverflow.com/questions/23351294/file-path-character-limit-error-in-windows-and-node-app 
https://github.com/joyent/node/issues/6960#issuecomment-46704998

I had to write a TFS build activity that could handle cleaning up node modules:

https://gist.github.com/colinbowern/bde095da68f6ee47d296

**Comment by steph643** [comment]
I get this same exception each time I try to install a package with `meteor add`. Is this currently the expected behavior? (just put my hands on your 0.9.0.1-rc3 and installed it with `LaunchMeteor09.exe`)

**Comment by sdarnell** [comment]
There was a problem with an earlier bundle that it didn't include that file (hybi\stream_reader.js). But I updated the bootstrap package with a fixed version. So if you have just installed with LaunchMeteor09.exe it should be OK.

However, the tools package on the package server still had it missing. But I can't see how you'd get that copy.

**Comment by sdarnell** [comment]
I'll be releasing a new version later today, and trying to get to the bottom of the issue or at least a workaround.

**Comment by steph643** [comment]
Ok. Will wait for your new version, try again and let you know.

**Comment by sdarnell** [comment]
I've released another build, windows:METEOR@0.9.0.1-rc4 which seems to be fixed now.
However, because of the bug you'll need to delete your `%LOCALAPPDATA%\.meteor` directory and re-run LaunchMeteor09.exe.

**Comment by steph643** [comment]
It works! I was able to install all my packages, except one (will create another ticket).
